src/workerd/io/worker.c++

```diff
@@ -1602,7 +1602,8 @@ Worker::Worker(kj::Own<const Script> scriptParam,
                 }
                 KJ_CASE_ONEOF(limitError, kj::Exception) { }
               }
-            }
+            } else {}
```

Comment:
These change seems unrelated. Can you remove it?

Reply:
This fixes the one fatal warning we have currently

Reply:
What is the warning? The statement looks really superfluous to me, if there's no other way to handle this let's add a comment here

Reply:
-Wdangling-else on KJ_IF_SOME. I see that James fixed a bunch of these in the past.  The macro tries to block that warning but it's still coming through. Maybe we should look into that.


Thinking out loud a bit: 

The macro's basic structure is 

```c++
if (has value)
   if (init; false) { 
      // no op
   } else 
```

So a usage expands to something like

```c++
if (has value) 
    if (init; false) { 
       // no op
    } else {
       // here is my actual block when there is a value 
    }
```

Clang is not happy about that outer if which has no braces.

Reply:
I can't trigger it on its own so I'm thinking either a clang bug, or something glitching with the pragmas

Reply:
I spent way too much time on this - the problem is triggered if KJ_IF_SOME is passed as an argument to any other macro.

e.g.
```c++
#define USELESS(x) x
USELESS(
    KJ_IF_SOME(foo, bar) {
       // blah
    }
)
```

Reply:
For now I added a comment to explain why that else is there.